openslides_backend/action/mixins/import_mixins.py

```diff
@@ -177,5 +180,51 @@ def validate_instance(self, instance: Dict[str, Any]) -> None:
                             raise ActionException(
                                 f"Could not parse {entry[field]} expect boolean"
                             )
+                    elif type_ == "date":
+                        try:
+                            entry[field] = int(
+                                mktime(strptime(entry[field], "%Y-%m-%d"))
+                            )
+                        except Exception:
+                            raise ActionException(
+                                f"Could not parse {entry[field]} except date"
+                            )
+                    elif type_ == "string[]":
+                        try:
+                            entry[field] = json.loads("[" + entry[field] + "]")
+                        except Exception:
+                            raise ActionException(
+                                f"Could not parse {entry[field]} except string[]"
+                            )
 
         super().validate_instance(instance)
+
+
+class Lookup:
+    def __init__(
+        self,
+        datastore: DatastoreService,
+        collection: str,
+        names: List[str],
+        field: str = "name",
+    ) -> None:
+        if not names:
+            self.name_to_id: Dict[str, Optional[int]] = {}
+        else:
+            self.name_to_id = {
+                entry[field]: entry["id"]
+                for entry in datastore.filter(
+                    collection,
+                    Or(*[FilterOperator(field, "=", name) for name in names]),
+                    ["id", field],
+                ).values()
+            }
+
+    def check_duplicate(self, name: str) -> bool:
+        result = name in self.name_to_id
+        if not result:
+            self.name_to_id[name] = None
+        return result
+
+    def get_id_by_name(self, name: str) -> Optional[int]:
```

Comment:
Isn't really useful anymore

Reply:
It is still useful.

Reply:
In case, that he found an id, the action needs to get the id. I think, this is with modification useful.